### DISCUSSION TITLE
Module object for pyimod02_importers is NULL! - ModuleNotFoundError: No module named 'pathlib'

### DISCUSSION THREAD

**Discussion by karam-jaber-intel** [pyinstaller/pyinstaller#8608] [Help]
I'm using miniforge conda package manager to build my environment . within it i have the following componetns:
Python 3.10.14
PyInstaller 6.8.0

When i freeze my app and then run it i get the following error :

```
[3594] Module object for pyimod02_importers is NULL!
Traceback (most recent call last):
  File "PyInstaller/loader/pyimod02_importers.py", line 22, in <module>
ModuleNotFoundError: No module named 'pathlib'
```

here is the log from pyinstaller:

```
Packaging using PyInstaller
549 INFO: PyInstaller: 6.8.0, contrib hooks: 2024.7
549 INFO: Python: 3.10.14 (conda)
553 INFO: Platform: Linux-5.14.21-150400.24.111-default-x86_64-with-glibc2.27
553 INFO: Python environment: /workspace/t/workspace/intelpython
562 INFO: Module search paths (PYTHONPATH):
['/workspace/t/workspace',
 '/workspace/t/workspace/intelpython/lib/python310.zip',
 '/workspace/t/workspace/intelpython/lib/python3.10',
 '/workspace/t/workspace/intelpython/lib/python3.10/lib-dynload',
 '/workspace/t/workspace/intelpython/lib/python3.10/site-packages',
 '/workspace/t/workspace',
 '/workspace/t/workspace']
781 INFO: Appending 'datas' from .spec
796 INFO: checking Analysis
828 INFO: checking PYZ
841 WARNING: Ignoring version information; supported only on Windows!
842 WARNING: Ignoring icon; supported only on Windows and macOS!
901 INFO: checking PKG
923 INFO: Bootloader /workspace/t/intelpython/lib/python3.10/site-packages/PyInstaller/bootloader/Linux-64bit-intel/run
924 INFO: checking EXE
936 INFO: Rebuilding EXE-00.toc because mfit missing
936 INFO: Building EXE from EXE-00.toc
936 INFO: Copying bootloader EXE to /workspace/t/workspace/dist/mfit
937 INFO: Appending PKG archive to custom ELF section in EXE
1015 INFO: Building EXE from EXE-00.toc completed successfully.
```

Any idea what could cause this ?

I tried to use the python and then run "import pathlib" and it did work . Its part of python's standard library.

**Comment by rokm** [comment]
>```
>[3594] Module object for pyimod02_importers is NULL!
>Traceback (most recent call last):
>  File "PyInstaller/loader/pyimod02_importers.py", line 22, in <module>
>ModuleNotFoundError: No module named 'pathlib'
>```

Hmm, we [aren't importing `pathlib` on line 22 in `pymod02_importers` in v6.8.0](https://github.com/pyinstaller/pyinstaller/blob/v6.8.0/PyInstaller/loader/pyimod02_importers.py), and we haven't been doing that since https://github.com/pyinstaller/pyinstaller/commit/9bc447f29bbcd164a64f2d5dd3d38ecbcabf9ea2, which was released with v6.0.0.

So my first guess would be that your current build is reusing cached files from earlier builds made with v5.x. Can you try performing a clean build, i.e., add `--clean` to your PyInstaller command?

**Reply by Bllose** [reply]
I've encountered the same issue. My Python version is 3.9, and I don't have network access, so I need to install everything offline. I'm very certain that there's only one Python version on my local machine. Also, all the dependent packages are in order. I've tried using two versions of PyInstaller, 3.7 and 3.10, but still got the same problem.
Adding --clean is not work for me, by the way.

My error information as below:
C:\Windows\System32\cmd.exe
Microsoft Windows [Version 10.0.19044.5608]
(c) Microsoft Corporation. All rights reserved.
 
D:\workplace\PyTools\AutoEmail\dist>auto_email.exe
[604] Module object for pymimod02_import ers is NULL!
Traceback (most recent call last):
File "PyInstaller\loader\pyimod02_import ers.py", line 22, in 
File "", line 259, in load_module
File "pathlib.py", line 13, in 
File "", line 259, in load_module
File "urllib\parse.py", line 39, in 
ModuleNotFoundError: No module named 'ipaddress'
Traceback (most recent call last):
File "PyInstaller\loader\pyiboot01_bootstrap.py", line 17, in 
ModuleNotFoundError: No module named 'pyimod02_import ers'
[604] Failed to execute script 'pyiboot01_bootstrap' due to unhandled exception!
 
:-(

**Reply by rokm** [reply]
>  `ModuleNotFoundError: No module named 'ipaddress'`

This looks like #9094. If you are using old PyInstaller 5.x series, it won't work with python >= 3.9.21 and >= 3.10.16.
